src/test/java/com/flab/bbt/auth/service/AuthServiceTest.java

```diff
@@ -0,0 +1,100 @@
+package com.flab.bbt.auth.service;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.flab.bbt.exception.CustomException;
+import com.flab.bbt.user.domain.User;
+import com.flab.bbt.user.repository.UserRepository;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AuthServiceTest {
+
+    @InjectMocks
+    private AuthService authService;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private PasswordEncrypter passwordEncrypter;
+
+    User user;
+
+    @BeforeEach
+    public void setUp() {
+        user = User.builder()
+            .email("test@test.com")
+            .password("encryptedPassword")
+            .name("test")
+            .phoneNo("01012341234")
+            .build();
+    }
+
+    @Test
+    @DisplayName("회원가입에 성공한다.")
+    void signUpTest() {
+        // given
+        when(userRepository.findByEmail(anyString())).thenReturn(Optional.empty());
+        when(userRepository.save(any(User.class))).thenReturn(user);
+
+        //when
+        authService.signUp(user);
+
+        //then
+        verify(userRepository).save(user);
+    }
+
+    @Test
+    @DisplayName("이메일 중복 시 회원가입 실패하며 예외가 발생한다.")
+    void signUpWithDuplicatedEmailUserTest() {
+        // given
+        when(userRepository.findByEmail(anyString())).thenReturn(Optional.ofNullable(user));
+
+        // when
+        CustomException e = assertThrows(CustomException.class,
+            () -> authService.signUp(user));
+
+        // then
+        assertThat(e.getErrorCode().getMessage()).isEqualTo("이메일이 중복되었습니다.");
```

Comment:
오류문구가 변경되면 테스트가 모두깨지고, 하나씩 변경해주어야 합니다.
오타가 날 여지도 있습니다.

아래처럼 한다면 "`ErrorCode`에 명시된 메세지가 내려가야 한다" 라는 검증을 할 수 있습니다
```suggestion
        assertThat(e.getErrorCode().getMessage()).isEqualTo(ErrorCode.USER_NOT_FOUND.getMessage());
```

Reply:
감사합니다! 수정하여 반영했습니다.